src/main/java/com/rabbitmq/client/RpcClient.java

```diff
@@ -79,12 +80,14 @@
         }
     };
 
+    public static Supplier<String> DEFAULT_CORRELATION_ID_GENERATOR = new IncrementingCorrelationIdGenerator("");
```

Comment:
As-is this static instance may be used by more than one `RpcClient` instance. In that case `Supplier#get` may be invoked concurrently, as the `synchronized (_continuationMap)` block only ensures exclusive access for `doCall` invocations on the same client. This is a problem because as-is `IncrementingCorrelationIdGenerator` is not thread-safe.

One option would be to replace `int _correlationId` with an `AtomicInteger` (or perhaps more appropriately, an `AtomicLong`.)

Reply:
Good catch. This should not be static. Will remove that.